examples/connext_dds/xml_application_env/c++11/xml_application_env.cpp

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ RTI grants Licensee a license to use, modify, compile, and create derivative
+ works of the Software.  Licensee has the right to distribute object form only
+ for use with RTI products.  The Software is provided "as is", with no warranty
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software.  RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ ******************************************************************************/
+
+#include <chrono>
+#include <cstdlib>
+#include <dds/dds.hpp>
+#include <iostream>
+#include <rti/rti.hpp>
+
+using namespace dds::core;
+using namespace dds::core::xtypes;
+using namespace dds::domain;
+using namespace dds::pub;
+using namespace dds::sub;
+using namespace rti::core;
+
+const std::string QOS_URL = "file://application.xml";
+const std::string PARTICIPANT_NAME = "domain_participant_library::participant";
+const std::string WRITER_NAME = "publisher::writer";
+const std::string READER_NAME = "subscriber::reader";
+const std::string KVP_TYPE_NAME = "KeyValuePair";
+
+class MyDataReaderListener : public NoOpDataReaderListener<DynamicData> {
+    virtual void on_data_available(DataReader<DynamicData> &reader)
+    {
+        LoanedSamples<DynamicData> samples = reader.take();
+        for (LoanedSamples<DynamicData>::iterator sampleIt = samples.begin();
+             sampleIt != samples.end();
+             ++sampleIt) {
+            // If the reference we get is valid data, it means we have actual
+            // data available, otherwise we received metadata.
+            if (sampleIt->info().valid()) {
```

Comment:
We could modify this comment slightly. You always receive metadata as part of the sample info. When valid = false, what you receive are changes on the dds::sub::status::InstanceState of the instance.

See https://community.rti.com/static/documentation/connext-dds/current/doc/api/connext_dds/api_cpp2/classdds_1_1sub_1_1SampleInfo.html#af77d354c564302067f707013f3053a2b

Reply:
This wording came directly from listeners of which it appears in three separate files (eg listeners/c++03/listeners_subscriber.cxx line 215).  

I suggest we just leave it for this commit, and if you would like it re-worded then create an issue to change it everywhere.

Reply:
I agree with you, but that comment was taken directly from the listeners example.  In fact it appears 3 times in that example (eg listeners/c++03/listeners_subscriber.cxx:215).  I suggest we just leave this as it is, and if you would like better wording then create a ticket and address them all.

Reply:
I agree.